### PR TITLE
fix(3235): Fetch parentEvent meta when create Event

### PR DESCRIPTION
**Pull request by kumada626**
## Context

When restart build was failed in `sd-setup-init` step and restart build from that build, meta is missing at that time.
So we fix this behavior by fetching parent event meta when creating event.

## Objective

Fetch parent event meta when creating event.

## References

#3225 

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Review by y-oksaku — APPROVED** [review]

**Review by yakanechi — APPROVED** [review]

File: plugins/events/create.js [hunk 1]
```diff
@@ -220,15 +220,24 @@ module.exports = () => ({
...
+                    mergedParameters = parentEvent.meta.parameters;
+                }
+
+                if (Object.keys(payload.meta).length === 0) {
+                    payload.meta = { ...parentEvent.meta };
+                }
+
+                if (Object.keys(mergedParameters).length > 0) {
+                    payload.meta.parameters = mergedParameters;
                 }
```
Comment (VonnyJap) [inline plugins/events/create.js:241]: what about make this more simplified?
```suggestion
                // Merge parameters if they exist in the parent event and not in the payload
                if (!payload.meta.parameters && parentEvent.meta && parentEvent.meta.parameters) {
                    payload.meta.parameters = parentEvent.meta.parameters;
                }
                
                // Copy meta from parent event if payload.meta is empty
                if (Object.keys(payload.meta).length === 0) {
                    payload.meta = { ...parentEvent.meta };
                }
```

**Review by VonnyJap — CHANGES_REQUESTED** [review]
please check my suggestion

File: plugins/events/create.js (hunk 1 above)
Comment (kumada626) [reply plugins/events/create.js:241]: If payload(RestartedEevnt) only has parameters in meta, I think your suggested method will not fetch parentEvent meta (and neither will my method).
I will have fixed this bug.

**Comment by VonnyJap** [comment]
@kumada626 - can you also add couple of functional tests 
https://github.com/screwdriver-cd/screwdriver/blob/master/features/metadata.feature
Changes to metadata behavior can be crucial and we should add some functional tests to make sure that it is working as intended if possible.

**Comment by kumada626** [comment]
@VonnyJap Since this change does not alter normal behavior, it would be difficult to add it as a functional test.
This change is to fix the case of a restart from a build that failed in setup-init of a build due to some temporary environmental factors, so if we can guarantee that the normal behavior has not changed, there should be no problem.

**Review by VonnyJap — APPROVED** [review]

**Pull request merged**